File: frontend/src/stores/analytics.js
import { defineStore } from 'pinia'
import { ref } from 'vue'
import { analyticsAPI } from '../utils/api'

export const useAnalyticsStore = defineStore('analytics', () => {
  const dashboardStats = ref(null)
  const engagementTrends = ref([])
  const platformStats = ref([])
  const topPosts = ref([])
  const demographics = ref(null)
  const loading = ref(false)
  const error = ref(null)

  const fetchDashboardStats = async () => {
    loading.value = true
    error.value = null
    try {
      const response = await analyticsAPI.getDashboardStats()
      dashboardStats.value = response.data
    } catch (err) {
      error.value = err.message
      console.error('Error fetching dashboard stats:', err)
    } finally {
      loading.value = false
    }
  }

  const fetchEngagementTrends = async (days = 30) => {
    loading.value = true
    error.value = null
    try {
      const response = await analyticsAPI.getEngagementTrends(days)
      engagementTrends.value = response.data
    } catch (err) {
      error.value = err.message
      console.error('Error fetching engagement trends:', err)
    } finally {
      loading.value = false
    }
  }

  const fetchPlatformStats = async () => {
    loading.value = true
    error.value = null
    try {
      const response = await analyticsAPI.getPlatformStats()
      platformStats.value = response.data
    } catch (err) {
      error.value = err.message
      console.error('Error fetching platform stats:', err)
    } finally {
      loading.value = false
    }
  }

  const fetchTopPosts = async (limit = 10) => {
    loading.value = true
    error.value = null
    try {
      const response = await analyticsAPI.getTopPosts(limit)
      topPosts.value = response.data
    } catch (err) {
      error.value = err.message
      console.error('Error fetching top posts:', err)
    } finally {
      loading.value = false
    }
  }

  const fetchDemographics = async (accountId = null) => {
    loading.value = true
    error.value = null
    try {
      const response = await analyticsAPI.getDemographics(accountId)
      demographics.value = response.data
    } catch (err) {
      error.value = err.message
      console.error('Error fetching demographics:', err)
    } finally {
      loading.value = false
    }
  }

  return {
    dashboardStats,
    engagementTrends,
    platformStats,
    topPosts,
    demographics,
    loading,
    error,
    fetchDashboardStats,
    fetchEngagementTrends,
    fetchPlatformStats,
    fetchTopPosts,
    fetchDemographics
  }
})

File: backend/routers/analytics.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session
from typing import List
from database import get_db
from services.analytics_service import AnalyticsService
from schemas import DashboardStats, EngagementTrend, PlatformStats

router = APIRouter(prefix="/api/analytics", tags=["analytics"])


@router.get("/dashboard", response_model=DashboardStats)
def get_dashboard_stats(db: Session = Depends(get_db)):
    """Get overall dashboard statistics"""
    return AnalyticsService.get_dashboard_stats(db)


@router.get("/trends", response_model=List[EngagementTrend])
def get_engagement_trends(days: int = 30, db: Session = Depends(get_db)):
    """Get engagement trends over specified days"""
    if days < 1 or days > 365:
        raise HTTPException(status_code=400, detail="Days must be between 1 and 365")
    return AnalyticsService.get_engagement_trends(db, days)


@router.get("/platforms", response_model=List[PlatformStats])
def get_platform_stats(db: Session = Depends(get_db)):
    """Get statistics by platform"""
    return AnalyticsService.get_platform_stats(db)


@router.get("/top-posts")
def get_top_posts(limit: int = 10, db: Session = Depends(get_db)):
    """Get top performing posts"""
    if limit < 1 or limit > 100:
        raise HTTPException(status_code=400, detail="Limit must be between 1 and 100")
    return AnalyticsService.get_top_posts(db, limit)


@router.get("/demographics")
def get_audience_demographics(account_id: int = None, db: Session = Depends(get_db)):
    """Get audience demographics"""
    return AnalyticsService.get_audience_demographics(db, account_id)
